metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/extractor/TaxonomyExtractorFromReferenceSequence.java

```diff
@@ -0,0 +1,46 @@
+/*
+ *
+ * Copyright 2019 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package uk.ac.ebi.ampt2d.metadata.importer.extractor;
+
+import uk.ac.ebi.ampt2d.metadata.persistence.entities.Taxonomy;
+import uk.ac.ebi.ampt2d.metadata.persistence.repositories.TaxonomyRepository;
+import uk.ac.ebi.ena.sra.xml.AssemblyType;
+
+public class TaxonomyExtractorFromReferenceSequence {
```

Comment:
I can't _yet_ rename it to `TaxonomyExtractor` because this name is taken by a dummy taxonomy generator, which will only be removed when we merge #54.

As for the class structure:
* The taxonomy extraction & persistence was originally implemented as a separate class;
* Then, following review comments and changing upstream class structure, moved inside the ReferenceSequence persistence logic;
* Then moved to a separate class _again_ as per @selvaebi's suggestion.

Currently it is modelled after `PublicationExtractorFromStudy` and similar other classes. Could you please elaborate on how I should change it and where to move its code?

Reply:
I guess my concern is due to the fact that this class doesn't really deal with a reference sequence at any point, just with taxonomies, and that is why I see it been closer to `StudyRepository`. If that was the case, we would need to replicate the `StudyRepositoryCustom(Impl)` class hierarchy, the `findOrCreate` method in the implementation instead of the top interface declaration.

Reply:
@cyenyxe I see your point now. I suppose the best way would be to implement a `TaxonomyRepositoryCustom` with  a `findOrCreate` method and to simply call that at the appropriate point during the reference sequence import